packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.h

```diff
@@ -163,6 +168,21 @@ typedef void (^RCTHostDidReceiveJSErrorStackBlock)(
  */
 @property (nonatomic, nullable) RCTHostDidReceiveJSErrorStackBlock hostDidReceiveJSErrorStackBlock;
 
+/**
+ * Called when the `RCTInstance` will start loading the bundle.
+ * @parameter: host - The `RCTHost` that owns the `RCTInstance`.
+ * @parameter: loadCallback
+ */
+@property (nonatomic, nullable) RCTHostLoadSourceBlock loadSourceForHost;
+
+/**
+ * Called when the `RCTInstance` will start loading the bundle.
+ * @parameter: host - The `RCTHost` that owns the `RCTInstance`.
+ * @parameter: onProgress - block called with progress updates.
+ * @parameter: loadCallback
+ */
+@property (nonatomic, nullable) RCTHostLoadSourceWithProgressBlock loadSourceWithProgressForHost;
```

Comment:
We can add these to a protocol `RCTCustomBundlerProvider`, mark them as optional and make the `RCTAppDelegate` conform to the protocol. In this way, you don't have to provide a default implementation in the RCTAppDelegate.